### PR TITLE
chore: post to mend slack when labeled

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Post to Mend Slack whenever a Discussion is labeled with `mend-app`

## Context

A way for us to alert the relevant team to user-impacting issues in the app

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/mend-slack.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: 'Mend Slack Notifier'
+
+on:
+  discussion:
+    types: [labeled]
+
+permissions:
+  discussions: read
```
Comment (Shegox) [inline .github/workflows/mend-slack.yml:8]: This is probably not even needed, but doesn’t really hurt either.

File: .github/workflows/mend-slack.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    if: contains(github.event.label.name, 'mend-app')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Post to Slack
+        id: slack
+        uses: slackapi/slack-github-action@v1.24.0
+        with:
+          channel-id: 'C05NLTMGCJC'
+          # For posting a simple plain text message
+          slack-message: "${{ github.event.discussion.title }}\n${{ github.event.discussion.html_url }}"
```
Comment (Shegox) [inline .github/workflows/mend-slack.yml:21]:
```suggestion
          slack-message: "<${{ github.event.discussion.html_url }}|${{ github.event.discussion.title }}>"
```

This syntax should put the link behind the discussion URL if desired: https://api.slack.com/reference/surfaces/formatting#linking-urls

**Review by Shegox — APPROVED** [review]
Looks right to me and should work.

**Review by viceice — DISMISSED** [review]
looks good

File: .github/workflows/mend-slack.yml (hunk 1 above)
Comment (viceice) [reply .github/workflows/mend-slack.yml:8]: this ensures that we only have this limited permission

**Pull request merged**